src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +258,135 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
+        } else {
+            analyzerObj = parseContext.mapperService().analysisService().analyzer(analyzer);
+            if (analyzerObj == null) {
+                throw new IllegalArgumentException("no analyzer found for [" + analyzer + "]");
+            }
+        }
+        
+        Occur highFreqOccur = highFreqOperator.toMustOrShouldClause();
+        Occur lowFreqOccur = lowFreqOperator.toMustOrShouldClause();
+
+        ExtendedCommonTermsQuery query = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, cutoffFrequency, disableCoord, mapper);
+        parseQueryString(query, text, field, analyzerObj, lowFreqMinimumShouldMatch, highFreqMinimumShouldMatch);
+
+        query.setBoost(boost);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+    
+    static final void parseQueryString(ExtendedCommonTermsQuery query, Object queryString, String field, Analyzer analyzer, 
+                                         String lowFreqMinimumShouldMatch, String highFreqMinimumShouldMatch) throws IOException {
+        // Logic similar to QueryParser#getFieldQuery
+        int count = 0;
+        try (TokenStream source = analyzer.tokenStream(field, queryString.toString())) {
+            source.reset();
+            CharTermAttribute termAtt = source.addAttribute(CharTermAttribute.class);
+            BytesRefBuilder builder = new BytesRefBuilder();
+            while (source.incrementToken()) {
+                // UTF-8
+                builder.copyChars(termAtt);
+                query.add(new Term(field, builder.toBytesRef()));
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            throw new IllegalArgumentException("common terms query could not analyze query string: [" + queryString + "]");
+        }
+        query.setLowFreqMinimumNumberShouldMatch(lowFreqMinimumShouldMatch);
+        query.setHighFreqMinimumNumberShouldMatch(highFreqMinimumShouldMatch);
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (Strings.isEmpty(this.fieldName)) {
+            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (this.text == null || this.text.toString().isEmpty()) {
```

Comment:
I think I would only leave the null check here, not sure I'd call toString on this object, that is what we did in term query too.